src/clients/c++/request.h

```diff
@@ -329,6 +329,22 @@ class InferContext {
     /// \param input The vector holding tensor string values.
     /// \return Error object indicating success or failure.
     virtual Error SetFromString(const std::vector<std::string>& input) = 0;
+
+    /// Set tensor values for this input by reference into a shared memory
+    /// region. The values are not copied and so the shared memory region and
+    /// its contents must not be modified or destroyed until this input is no
+    /// longer needed (that is until the Run() call(s) that use the input have
+    /// For batched inputs, all tensor values must be contiguous in a single
```

Comment:
"This function must be called a single time for an input that is using shared memory. For batched inputs, the tensor values for the entire batch must be contiguous..."